From 6121bcb23b0b306f0babc0c254952f8fc14ff95d Mon Sep 17 00:00:00 2001
From: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Date: Tue, 26 Nov 2013 11:11:17 +0000
Subject: [PATCH 056/351] mfd: arizona: fix potential bad pointer when
 printing chip ID

If a Florida-type codec is found but the driver was not
registered as a known Florida-type part the previous code
would use an uninitialized pointer when logging that the
codec was registered. This change fixes that.

Change-Id: If85df6b00d5f4a719851f28c192322126245c9f6
Signed-off-by: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/mfd/arizona-core.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/arizona-core.c b/drivers/mfd/arizona-core.c
index dc91bed..3ce3ee0 100644
--- a/drivers/mfd/arizona-core.c
+++ b/drivers/mfd/arizona-core.c
@@ -865,7 +865,7 @@ static const struct mfd_cell wm8997_devs[] = {
 int arizona_dev_init(struct arizona *arizona)
 {
 	struct device *dev = arizona->dev;
-	const char *type_name;
+	const char *type_name = "Unknown";
 	unsigned int reg, val;
 	int (*apply_patch)(struct arizona *) = NULL;
 	int ret, i;
@@ -1051,6 +1051,8 @@ int arizona_dev_init(struct arizona *arizona)
 			dev_err(arizona->dev, "Florida codec registered as %d\n",
 				arizona->type);
 			arizona->type = WM8280;
+			type_name = "Florida";
+			revision_char = arizona->rev + 61;
 			break;
 		}
 		apply_patch = florida_patch;
-- 
1.7.9.5

